packages/inline-edit/src/InlineEdit.tsx

```diff
@@ -127,14 +127,24 @@ const InlineEdit = forwardRef<HTMLInputElement, InlineEditProps>(
       </>
     );
 
+    const inputProps = {
+      ref: mergeRefs(inputRef, ref),
+      defaultValue,
+      onKeyDown: handleKeyDown,
+      'aria-label': ariaLabel,
+    };
+
+    const inputChildren = renderInput.props.children;
+
     const input = cloneElement(
       renderInput,
-      mergeProps(renderInput.props, {
-        ref: mergeRefs(inputRef, ref),
-        defaultValue,
-        onKeyDown: handleKeyDown,
-        'aria-label': ariaLabel,
-      })
+      mergeProps(renderInput.props, inputChildren ? {} : inputProps),
+      inputChildren &&
+        Children.map(inputChildren, (child) =>
+          ['TextField', 'TextArea'].includes(child.type.displayName)
```

Comment:
Why do we need to reach for the display name here? Is reading `displayName` this way reliable post-bundling, etc…

Reply:
@apucacao, I'm guessing this is to check that, if it's an expected Launchpad-based React element, then we merge its props with the `inputProps` we need. I don't know anything about bundling effects on this.

Reply:
Yeah, I figured.

I was wondering why `child.type === TextField` wouldn't work. We'd need to import `TextField`.

Reply:
LP component display names aren't minified in the product but just to be safe I can check for the type to match the component itself.